### ISSUE TITLE
Limited compatibility with spongeforge

### ISSUE THREAD

**Issue by ShiratsuYudachi** [ValkyrienSkies/Valkyrien-Skies#418]
Tested with:

Valkyrien Skies Control 1.0.1
Valkyrien Skies 1.0.0
Valkyrien Skies World 1.0.0
forge-1.12.2-14.23.5.2838
spongeforge-1.12.2-2838-7.3.0
*no other mods or sponge plugins are installed

the ship can be assembled and controled by usinng captain chair normally
but other blocks include the compressor, engines, and even stabilizer and compacted valkyrium, are no longer affect to the ship except they have already activated before assemble ship. 
also, all blocks placed after assemble ship will disappear when disassemble ship.
and in F3+B view, the weight of the airship will not changed by place or break blocks after assemble ship
I think it is some problem that relating to update.

a part from this, it will cause an error when performing /help
(the server will not be crushed)

> help
[21:19:38] [Server thread/INFO] [minecraft/DedicatedServer]: Error occurred while executing command: net/minecraft/client/multiplayer/WorldClient
[21:19:38] [Server thread/ERROR] [Sponge]: Error occurred while executing command 'help' for source DedicatedServer: net/minecraft/client/multiplayer/WorldClient
java.lang.NoClassDefFoundError: net/minecraft/client/multiplayer/WorldClient
        at java.lang.Class.getDeclaredConstructors0(Native Method) ~[?:1.8.0_261]
        at java.lang.Class.privateGetDeclaredConstructors(Unknown Source) ~[?:1.8.0_261]
        at java.lang.Class.getConstructor0(Unknown Source) ~[?:1.8.0_261]
        at java.lang.Class.getDeclaredConstructor(Unknown Source) ~[?:1.8.0_261]
        at org.valkyrienskies.mod.common.command.framework.VSCommandFactory.create(VSCommandFactory.java:24) ~[VSCommandFactory.class:?]
        at picocli.CommandLine$Model$CommandReflection.initSubcommands(CommandLine.java:9585) ~[CommandLine$Model$CommandReflection.class:?]
        at picocli.CommandLine$Model$CommandReflection.extractCommandSpec(CommandLine.java:9521) ~[CommandLine$Model$CommandReflection.class:?]
        at picocli.CommandLine$Model$CommandSpec.forAnnotatedObject(CommandLine.java:5116) ~[CommandLine$Model$CommandSpec.class:?]
        at picocli.CommandLine.<init>(CommandLine.java:223) ~[CommandLine.class:?]
        at org.valkyrienskies.mod.common.command.framework.VSCommandBase.func_71518_a(VSCommandBase.java:93) ~[VSCommandBase.class:?]
        at org.spongepowered.common.command.MinecraftCommandWrapper.getHelp(MinecraftCommandWrapper.java:204) ~[MinecraftCommandWrapper.class:1.12.2-2838-7.3.0]
        at org.spongepowered.common.command.MinecraftCommandWrapper.getShortDescription(MinecraftCommandWrapper.java:199) ~[MinecraftCommandWrapper.class:1.12.2-2838-7.3.0]
        at org.spongepowered.common.command.SpongeCommandFactory.createDescription(SpongeCommandFactory.java:1089) ~[SpongeCommandFactory.class:1.12.2-2838-7.3.0]
        at org.spongepowered.common.command.SpongeCommandFactory.lambda$null$28(SpongeCommandFactory.java:926) ~[SpongeCommandFactory.class:1.12.2-2838-7.3.0]
        at java.util.stream.ReferencePipeline$3$1.accept(Unknown Source) ~[?:1.8.0_261]
        at java.util.TreeMap$KeySpliterator.forEachRemaining(Unknown Source) ~[?:1.8.0_261]
        at java.util.stream.AbstractPipeline.copyInto(Unknown Source) ~[?:1.8.0_261]
        at java.util.stream.AbstractPipeline.wrapAndCopyInto(Unknown Source) ~[?:1.8.0_261]
        at java.util.stream.ReduceOps$ReduceOp.evaluateSequential(Unknown Source) ~[?:1.8.0_261]
        at java.util.stream.AbstractPipeline.evaluate(Unknown Source) ~[?:1.8.0_261]
        at java.util.stream.ReferencePipeline.collect(Unknown Source) ~[?:1.8.0_261]
        at org.spongepowered.common.command.SpongeCommandFactory.lambda$createHelpCommand$29(SpongeCommandFactory.java:926) ~[SpongeCommandFactory.class:1.12.2-2838-7.3.0]
        at org.spongepowered.api.command.spec.CommandSpec.process(CommandSpec.java:388) ~[CommandSpec.class:1.12.2-2838-7.3.0]
        at org.spongepowered.common.command.SpongeCommandDispatcher.process(SpongeCommandDispatcher.java:355) ~[SpongeCommandDispatcher.class:1.12.2-2838-7.3.0]
        at org.spongepowered.common.command.SpongeCommandManager.process(SpongeCommandManager.java:360) [SpongeCommandManager.class:1.12.2-2838-7.3.0]
        at net.minecraft.command.ServerCommandManager.func_71556_a(SourceFile:665) [dh.class:?]
        at net.minecraft.server.dedicated.DedicatedServer.func_71333_ah(DedicatedServer.java:433) [nz.class:?]
        at net.minecraft.server.dedicated.DedicatedServer.func_71190_q(DedicatedServer.java:398) [nz.class:?]
        at net.minecraft.server.MinecraftServer.func_71217_p(MinecraftServer.java:668) [MinecraftServer.class:?]
        at net.minecraft.server.MinecraftServer.run(MinecraftServer.java:526) [MinecraftServer.class:?]
        at java.lang.Thread.run(Unknown Source) [?:1.8.0_261]
Caused by: java.lang.ClassNotFoundException: net.minecraft.client.multiplayer.WorldClient
        at net.minecraft.launchwrapper.LaunchClassLoader.findClass(LaunchClassLoader.java:101) ~[launchwrapper-1.12.jar:?]
        at java.lang.ClassLoader.loadClass(Unknown Source) ~[?:1.8.0_261]
        at java.lang.ClassLoader.loadClass(Unknown Source) ~[?:1.8.0_261]
        ... 31 more
>

**Comment by xunto** [comment]
Yeah, happens to me too. Also it looks like vs can't notice sponge:
```
[16:53:56] [main/WARN] [FML]: ****************************************
[16:53:56] [main/WARN] [FML]: * Not applyingorg.valkyrienskies.mixin.spongepowered.common.network.MixinPacketUtil because Sponge isn't loaded!
[16:53:56] [main/WARN] [FML]: *  at org.valkyrienskies.mod.common.MixinLoadManager.shouldApplyMixin(MixinLoadManager.java:40)
[16:53:56] [main/WARN] [FML]: *  at org.spongepowered.asm.mixin.transformer.PluginHandle.shouldApplyMixin(PluginHandle.java:132)
[16:53:56] [main/WARN] [FML]: *  at org.spongepowered.asm.mixin.transformer.MixinInfo.shouldApplyMixin(MixinInfo.java:977)
[16:53:56] [main/WARN] [FML]: *  at org.spongepowered.asm.mixin.transformer.MixinInfo.readDeclaredTargets(MixinInfo.java:942)
[16:53:56] [main/WARN] [FML]: *  at org.spongepowered.asm.mixin.transformer.MixinInfo.<init>(MixinInfo.java:871)
[16:53:56] [main/WARN] [FML]: *  at org.spongepowered.asm.mixin.transformer.MixinConfig.prepareMixins(MixinConfig.java:713)...
[16:53:56] [main/WARN] [FML]: ****************************************
```
I also tried to just return `true` in method that checks for sponge and it didn't help. If I remeber correctly it lead to crash, so problem is probably deeper than that.

It was suggested in discord to downgrade sponge but I wasn't able to find correct version. While `spongeVersion = "1.12.2-2825-7.1.6"` (whatever it is) mentioned in `build.gradle` just crashed the game.

**Comment by ShiratsuYudachi** [comment]
in the Valkyrien Skies 1.1.1, causing error when performing /help is fixed. but other problems are still exist

**Comment by Tri0de** [comment]
Alright I've fixed this for SpongeForge 1.12.2-2825-7.1.6 in commit ed5f89535f2f572b46a1cdb833e960c676506a1b, so if you want to use sponge use that version. I haven't fixed it yet for the current recommended version, which is SpongeForge 1.12.2-2838-7.2.0.

**Comment by Tri0de** [comment]
Alright version 7.3.0 works as of commit febf7d6cd56af3318ae19a188bd599ccfc18d5db.
